### ISSUE TITLE
tslint doesn't report errors inside a function

### ISSUE THREAD

**Issue by tgroutars** [palantir/tslint#4401]
### Bug Report

- __TSLint version__: 5.12.0
- __TypeScript version__: 3.2.2
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
const foo = () => {
  const bar = baz();
};
```

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "error",
  "extends": ["tslint:latest"],
  "jsRules": {},
  "rules": {},
  "rulesDirectory": []
}
```

#### Actual behavior
No output and tslint exits succesfully

#### Expected behavior
Obviously there are several errors in this code, `baz` isn't defined for example

Is there something I'm missing?

**Comment by johnwiseheart** [comment]
`baz` not being defined in this case should be a `typescript` error.

If, for instance, you were to use this code: 
```ts
const foo = () => {
  const bar= baz();
};
```

You should see the `tslint` `whitespace` rule alert on the `=` sign.
